internal/controller/serverset/serverset.go

```diff
@@ -377,26 +457,23 @@ func (c *external) ensureVolumeClaim() error {
 	return nil
 }
 
-func (c *external) ensureServer(ctx context.Context, cr *v1alpha1.ServerSet, idx int) error {
+func (c *external) ensureServer(ctx context.Context, cr *v1alpha1.ServerSet, replicaIndex, version int) error {
 	c.log.Info("Ensuring Server")
 
-	name := getNameFromIndex(cr.Name, "server", idx)
-	ns := cr.Namespace
-	obj, err := c.getServer(ctx, name, ns)
+	resourceType := "server"
```

Comment:
we have a bunch of magic strings, please go ahead and create a `const.go` within our serverest package:

```
const (
    resourceServer = "server"
    resourceVolume = "volume" ... etc
)
```

This pattern reduces typos/mistakes